pkg/controller/database/rdsinstance.go

```diff
@@ -196,6 +198,14 @@ func (e *external) RestoreOrCreate(ctx context.Context, cr *v1beta1.RDSInstance,
 		if err != nil {
 			return awsclient.Wrap(err, errSnapshotRestoreFailed)
 		}
+	case "PointInTime":
+		if cr.Spec.ForProvider.RestoreFrom.PointInTime.SourceDBInstanceIdentifier == nil && cr.Spec.ForProvider.RestoreFrom.PointInTime.SourceDbiResourceID == nil && cr.Spec.ForProvider.RestoreFrom.PointInTime.SourceDBInstanceAutomatedBackupsArn == nil {
+			return errors.New(errPointInTimeRestoreSourceNotDefined)
+		}
```

Comment:
I don't know if it is better to give this custom error message early, or send the request to the AWS API and return that error.

Reply:
i would prefer early